### ISSUE TITLE
Add support for new CIAs

### ISSUE THREAD

**Issue by dirkwhoffmann** [dirkwhoffmann/virtualc64#423]
Currently, only the old CIAs are emulated.

**Comment by puleyo** [comment]
Thank you, the new CIA are the best

**Comment by Alessandro1970** [comment]
..but the two CIA 6526a or 6526 are quite identical...

_The difference seems to be that new CIA chip triggers interrupts 1 cycle before the old one does (it has nothing to do with reaching 0 vs. underflowing). The old CIA triggers at the cycle after the underflow,
and the new CIA triggers at the cycle of the underflow. The interrupt is delayed by one cycle._

**[cia6526.zip](https://github.com/dirkwhoffmann/virtualc64/files/2415559/cia6526.zip)**

**Comment by dirkwhoffmann** [comment]
Current status (Lorenz test suite): 

![Uploading Bildschirmfoto 2018-10-01 um 13.17.42.png…]()

**Comment by dirkwhoffmann** [comment]
Current progress (Lorenz test suite):

<img width="928" alt="bildschirmfoto 2018-10-01 um 22 04 02" src="https://user-images.githubusercontent.com/12561945/46294751-12d75380-c5c9-11e8-8a87-46966e280c92.png">

**Comment by dirkwhoffmann** [comment]
A new popup button "Hardware Configuration" has been added. It can be used to select one out of six hardware models (C64 PAL old, C64 PAL, C64 II PAL, C64 NTSC old, C64 NTSC, C64 II NTSC). All sub components (VIC, CIA, SID etc.) are then adjusted automatically to match the selected model.

<img width="562" alt="bildschirmfoto 2018-10-03 um 12 31 11" src="https://user-images.githubusercontent.com/12561945/46405348-4a004e80-c708-11e8-84ce-160305f6f56b.png">

**Comment by dirkwhoffmann** [comment]
V3.0 beta 1 can be downloaded from the website. The most critical issues (multi-threading performance issue, missing public key for the Sparkle updater) are fixed.

**Comment by dirkwhoffmann** [comment]
The "Hardware Configuration" popup button had a short life-time. I've replaced it with an enhanced "Restore Factory Settings" button which lets the user choose the C64 model to emulate. Seems much simpler and more reasonable to me.

<img width="914" alt="bildschirmfoto 2018-10-05 um 18 23 52" src="https://user-images.githubusercontent.com/12561945/46547433-2259e380-c8cc-11e8-8a68-e8e63c000ed3.png">

Furthermore, a "Emulate Timer B bug" checkbox has been added. It allows to alter the behavior of the old NMOS CIAs.
